### PR TITLE
Events

### PR DESCRIPTION
**Pull request by ryanlyrl**
Adds events.

* CRUD for events
* Users can favourite events

### CODE REVIEW

File: src/controllers/eventController.ts [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    const resourceRepository = getManager().getRepository(Resource);
+
+    let eventData = req.body;
+    if(!eventData.startTime) {
+        res.status(400).send("Missing start time");
+    }
+    eventData.startTime = new Date(eventData.startTime);
+    if(eventData.endTime) {
+        eventData.endTime = new Date(eventData.endTime);
+    }
```
Comment (dellisd) [inline src/controllers/eventController.ts:20]: Can these accept RFC3339 strings instead of unix timestamps?

**Review by dellisd — CHANGES_REQUESTED** [review]
Can the `getEvent` and `getAllEvents` endpoints return the start and end times as RFC3339 strings as well? Otherwise it looks good.

File: src/controllers/eventController.ts (hunk 1 above)
Comment (ryanlyrl) [reply src/controllers/eventController.ts:20]: The Date constructor can take in a bunch of different formats, including ISO8601 (which I think is very similar to RFC3339, if not the same?). This is an example of a time I tested with: `2020-01-21T06:00:00.000Z`.

About the returning, if I understand correctly, when the Date object is sent by Express, it gets turned into JSON, which calls Date's toJSON(): https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toJSON
That calls toISOString(), which is again, ISO8601.

Comment (dellisd) [reply src/controllers/eventController.ts:20]: Ok that sounds fine. We'll use ISO8601.

File: src/entity/Event.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    @IsDefined()
+    location!: string;
+
+    @Column()
+    @IsDefined()
+    locationName!: string;
+
+    @Column()
+    @IsDefined()
+    host!: string;
```
Comment (dellisd) [inline src/entity/Event.ts]: These should be nullable.

**Review by dellisd — CHANGES_REQUESTED** [review]

**Review by dellisd — APPROVED** [review]

**Pull request merged**